File: arena/src/problem_objects/array_read_swap.lua
local array = {}

return function(entries)
  array = entries
  return {
    read = function(index) return array[index] end,
    swap = function(a, b)
      if array[a] and array[b] then
         local tmp = array[a]
         array[a] = array[b]
         array[b] = tmp
      else
        error('tried to swap from an index that has no data')
      end
     end
  }
end

File: arena/entry_point/problem_object/src/array_read_swap.dart
class array_read_swap {

  var _array;

  array_read_swap(var array) {
    _array = array;
  }

  int read(index) {
    return _array[index];
  }

  void swap(index1, index2) {
    final tmp = _array[index1];
    _array[index1] = _array[index2];
    _array[index2] = tmp;
  }

  void length() {
    return _array.length;
  }

  void get array => _array;
}
